kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala

```diff
@@ -1652,6 +1652,12 @@ object KyuubiConf {
     .checkValues(EngineType.values.map(_.toString))
     .createWithDefault(EngineType.SPARK_SQL.toString)
 
+  val ENGINE_POOL_FIRST: ConfigEntry[Boolean] = buildConf("kyuubi.engine.pool.first")
```

Comment:
we shall change it to `kyuubi.engine.pool.ignoreSubdomain` and mark it as internal, so the primary users will not care about what's the relationship between engine pool and subdomain